internal/admission/validator.go

```diff
@@ -59,6 +60,7 @@ type KongHTTPValidator struct {
 	Logger                   logr.Logger
 	SecretGetter             kongstate.SecretGetter
 	ConsumerGetter           ConsumerGetter
+	Storer                   store.Storer
```

Comment:
Hmm, as I looked into it, it could indeed work for `Consumers`, but `Secrets` may not be ingested to the `Store` before being referred by another object (e.g. `Consumer`). I'm not sure if it's worth the effort to investigate and make sure it won't break in some unexpected way, to be honest. 🤷